controller/service.go

```diff
@@ -47,76 +52,92 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 		return true
 	}
 
-	// The assigned LB IP is the end state of convergence. If there's
+	// The assigned LB IP(s) is the end state of convergence. If there's
 	// none or a malformed one, nuke all controlled state so that we
 	// start converging from a clean slate.
-	if len(svc.Status.LoadBalancer.Ingress) == 1 {
-		lbIP = net.ParseIP(svc.Status.LoadBalancer.Ingress[0].IP)
-	}
-	if lbIP == nil {
-		c.clearServiceState(key, svc)
+	for i := range svc.Status.LoadBalancer.Ingress {
+		ip := svc.Status.LoadBalancer.Ingress[i].IP
+		if len(ip) != 0 {
+			lbIPs = append(lbIPs, net.ParseIP(ip))
+		}
 	}
 
-	// Clear the lbIP if it has a different ipFamily compared to the clusterIP.
-	// (this should not happen since the "ipFamily" of a service is immutable)
-	if (clusterIP.To4() == nil) != (lbIP.To4() == nil) {
+	lbIPsIPFamily := ipfamily.Unknown
+	if len(lbIPs) == 0 {
 		c.clearServiceState(key, svc)
-		lbIP = nil
+	} else {
+		lbIPsIPFamily, err = ipfamily.ForAddressesIPs(lbIPs)
+		if err != nil {
+			level.Info(l).Log("event", "clearAssignment", "reason", "nolbIPsIPFamily", "msg", "Failed to retrieve lbIPs family")
+			c.clearServiceState(key, svc)
```

Comment:
no need to clear in case of error.

Reply:
hmm code above when it return it does also clear so I followed the same ?

Reply:
oh, I looked at the first error occurrence here https://github.com/metallb/metallb/pull/1065/files#diff-e2b039cdee64b5555fd0499e1a75d9b5a42ac60c352a9680ccc9ebcac66cba42R134 and it wasn't, but there's another one later that clears.
It's fine this way I guess, no need to change.